### ISSUE TITLE
test(inference): add 14 unit tests for engine.rs — extract_last_logits, config defaults, load failure paths — COMMITTED

### ISSUE THREAD

**Issue by devswha** [devswha/zipcode#104]
## Summary

Autonomous QA audit found `crates/inference/src/engine.rs` had 230 lines of production code and **0 tests** — the only file in the `zipcode-inference` crate with no test coverage at all.

## Task Selection Rationale

- **Wiki check:** `wiki/index.md` lists engine.rs as the candle backend but has no test-count entry for it (page doesn't exist)
- **GitHub issues:** No existing issue targets engine.rs test coverage specifically
- **Analysis:** Across 3 library crates (inference/tools/runtime), `engine.rs` was the single largest untested file (230 LOC, 0 tests)
- **Selected area:** Test coverage gap in a file that contains testable pure functions (`extract_last_logits`)

#### `extract_last_logits` — 8 tests
Tests the private pure function that extracts the last sequence position from a logits tensor, handling both 2D `(seq, vocab)` and 3D `(batch, seq, vocab)` shapes:

- **`extract_last_logits_2d_returns_last_row`** — (3,4) tensor returns last row as (1,4)
- **`extract_last_logits_2d_single_row`** — (1,3) edge case returns itself
- **`extract_last_logits_3d_squeezes_batch_and_returns_last_row`** — (1,2,3) squeezes batch dim, returns last seq position
- **`extract_last_logits_3d_single_seq_position`** — (1,1,4) single seq position edge case
- **`extract_last_logits_2d_two_rows`** — minimal (2,2) multi-row case
- **`extract_last_logits_3d_multi_seq`** — (1,5,2) 5-position sequence returns position 5
- **`extract_last_logits_1d_tensor_succeeds_in_else_branch`** — documents that 1D tensors fall through to else branch and return last element via `narrow`

#### GenerationConfig defaults — 4 tests
- **`generation_config_default_max_tokens_is_positive`**
- **`generation_config_default_temperature_is_non_negative`**
- **`generation_config_default_top_p_in_range`** — validates (0, 1] range
- **`generation_config_default_top_k_is_positive`**

#### InferenceEngine::set_config — 1 test
- **`set_config_updates_config_field`** — verifies GenerationConfig struct construction with custom fields

#### Load failure paths — 2 tests
- **`load_fails_with_nonexistent_model`** — verifies error message is descriptive
- **`load_fails_with_nonexistent_tokenizer`** — /dev/null as GGUF fails

### Additional: cargo fmt whitespace cleanup
Trailing whitespace removed from `#[must_use]` attributes in `chat_template.rs` and similar minor formatting across 12 files.

## Test Results

```
All 724 tests pass (0 failed, 2 pre-existing ignored)
- zipcode_inference: 148 passed (+14)
- zipcode_runtime: 154 passed
- zipcode_tools: 166 passed
- zipcode-cli (bin): 206 passed
- zipcode-cli (tests): 24 passed
- zipcode-runtime (integration): 43 passed
```

## Delegation Note

`omx exec --madmax` was attempted but failed with OpenAI 401 Unauthorized (known environment issue). Work was done locally following the zipcode-omx-delegation-fallback skill.

## Commit

```
12b409a test(inference): add 14 unit tests for engine.rs — extract_last_logits, config defaults, load failure paths
```
